presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -453,18 +448,20 @@ public URI getRemoteTaskLocation()
     @Override
     public void start()
     {
-        try (SetThreadName ignored = new SetThreadName("HttpRemoteTask-%s", taskId)) {
-            // to start we just need to trigger an update
-            started.set(true);
-            scheduleUpdate();
-
-            taskStatusFetcher.start();
-            taskInfoFetcher.start();
-        }
+        taskEventLoop.execute(() -> {
+            try (SetThreadName ignored = new SetThreadName("HttpRemoteTask-%s", taskId)) {
```

Comment:
I would recommend not to override the thread name (here and in other places). It is easier to diagnose threading related issues when the threads are named consistently (e.g.: allowing you to find all threads belonging to a certain thread pool and all related events for it)